specs/protocol/holocene/derivation.md

```diff
@@ -1,13 +1,332 @@
+frame loading becomes simpler in the channel bank:
+
+- A first frame for a new channel starts a new channel as the staging channel.
+  - If there already is an open, non-completed staging channel, it is dropped and replaced by this
+  new channel. This is consistent with how the frame queue drops all frames of an non-closed channel
+  upon the arrival of a first frame for a new channel.
+- If the current channel is timed-out, but not yet pruned, and the incoming frame would be the next
+correct frame for this channel, the frame and channel are dropped, including all future frames for
+the channel that might still be in the frame queue. Note that the equivalent rule was already
+present pre-Holocene.
+
+## Span Batches
+
+Partial Span Batch Validity changes the atomic validity model of [Span Batches](../delta/span-batches.md).
+In Holocene, a span batch is treated as an optional stage in the derivation pipeline that sits
+before the batch queue, so that the batch queue pulls singular batches from this previous Span Batch
+stage. When encountering an invalid singular batch, it is dropped, as is the remaining span batch
+for consistency reasons. We call this _forwards-invalidation_. However, we don't
+_backwards-invalidate_ previous valid batches that came from the same span batch, as pre-Holocene.
+
+An implementation may choose to put the current span batch, from which singular batches are derived
+from, into a separate stage, or process it within the batch queue under the new rules.
+
+When a batch derived from the current staging channel is a singular batch, it is directly forwarded
+to the batch queue. Otherwise, it is set as the current span batch in the span batch stage. The
+following span batch validity checks are done, before singular batches are derived from it.
+Definitions are borrowed from the [original Span Batch specs](../delta/span-batches.md).
+
+- If the span batch _L1 origin check_ is not part of the canonical L1 chain, the span batch is
+invalid.
+- A failed parent check invalidates the span batch.
+- If `span_start.timestamp > next_timestamp`, the span batch is invalid, because we disallow gaps
+due to the new strict batch ordering rules.
+- If `span_end.timestamp < next_timestamp`, the span batch is invalid, as it doesn't contain any new
+batches (this would also happen if applying timestamp checks to each derived singular batch
+individually).
+- Note that we still allow span batches to overlap with the safe chain (`span_start.timestamp < next_timestamp`).
+
+If any of the above checks invalidate the span batch, it is dropped and the remaining channel from
+which the span batch was derived, is also immediately dropped (see also
+[Fast Channel Invalidation](#fast-channel-invalidation)).
+
+## Batch Queue
+
+The batch queue is also simplified in that batches are required to arrive strictly ordered, and any
+batches that violate the ordering requirements are immediately dropped, instead of buffered.
```

Comment:
If a batch arrives out of order, say due to a bug in the batcher, would this mean the safe chain would halt and not start again until the gap was filled?

Reply:
Exactly. This is intended. So the batcher can continuously monitor the sync status of the sequencer, and if its L1 inclusion block is past the last batcher transaction, but the safe chain hasn't advanced as expected, it should batch the non-safe unsafe section of the chain again. This is described in the batcher hardening section.